### ISSUE TITLE
[question] provide data via custom plugin and avoid edges (something like `site`)

### ISSUE THREAD

**Issue by egorovli** [gatsbyjs/gatsby#11335]
Hi 👋 I've setup a simple source plugin to provide locale object to the site (logic mainly taken from here https://www.gatsbyjs.org/docs/source-plugin-tutorial/). It basically reads a `yaml` file and creates a node with its content.

Although accessing the data is pretty awkward:

```js
graphql`
  query {
    locales: allLocale (limit: 1) {
      edges {
        node {
          ...LocaleEntries
        }
      }
    }
  }
`
```

and

```js
let locale = data.locales.edges[0].node
```

How can I provide this object avoiding all these edges and nodes, like the default `site` field on type `Query`, which does not have `allSite.edges[0].node`?

The plugin is as follows, just in case:

```js
let path = require('path')
let fs = require('fs')
let jsYaml = require('js-yaml')

let sourceNodes = (
  { actions, createNodeId, createContentDigest },
  configOptions
) => {
  let { createNode } = actions
  delete configOptions.plugins

  let filePath = path.join(configOptions.path, `${configOptions.locale}.yml`)
  let contents = fs.readFileSync(filePath, { encoding: 'utf-8' })
  let locale = jsYaml.load(contents)

  let nodeId = createNodeId(`locale-${locale.locale_language}`)
  let nodeContent = JSON.stringify(locale)
  let nodeData = Object.assign({}, locale, {
    id: nodeId,
    parent: null,
    children: [],
    internal: {
      type: 'Locale',
      content: nodeContent,
      contentDigest: createContentDigest(locale)
    }
  })

  createNode(nodeData)
}

module.exports = {
  sourceNodes
}
```

**Comment by KyleAMathews** [comment]
You can just directly require a yaml file into your components — e.g. `const localeData = require("../data/locale.yaml")`

**Comment by egorovli** [comment]
Woah. That was unexpected 😅 Thanks!

**Comment by egorovli** [comment]
I mean, nothing wrong with interpolation in a `require` call? Like:

```js
let locale = require(`../locales/${process.env.BUILD_LANG || 'en'}.yml`)
```

**Comment by KyleAMathews** [comment]
Oh kinda. Webpack gets weird with that. You'll want to use something called `require.context`.

But if you have more complex data querying requirements, webpack isn't suited for that. That's why GraphQL is in Gatsby so you can treat your data as a database and query it.

**Comment by KyleAMathews** [comment]
BTW, instead of writing your own source plugin, you can use https://www.gatsbyjs.org/packages/gatsby-source-filesystem/?=source-file https://www.gatsbyjs.org/docs/sourcing-from-the-filesystem/#sourcing-from-the-filesystem

**Comment by egorovli** [comment]
I tried that initially with `gatsby-transformer-yaml`, but I needed a way to pass `BUILD_LANG` env variables to the graphql query, and I found out it was impossible with `StaticQuery`. The original idea was to have a `Layout` component which receives the locale from graphql and passes it down via a context provider, but that is unfortunately not possible.

Anyway, the direct `require` call is way simpler, thanks for the hint!
